sdk-extensions/tracing-incubator/src/test/java/io/opentelemetry/sdk/extension/incubator/trace/LeakDetectingSpanProcessorTest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.incubator.trace;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+public class LeakDetectingSpanProcessorTest {
+
+  @Test
+  void garbageCollectedScope() {
+    List<Throwable> logs = new ArrayList<>();
```

Comment:
Could use `logunit`?

Reply:
I originally had logunit and @breedx-splk suggested doing it this way. I think the idea was that we could easily make providing the consumer part of the public API for use-cases that didn't want to just log, but wanted to gather the raw data and do something other than logging with it.

Honestly, I'm not sure it ends up mattering all that much, but will switch back to making assertions about logs if you think that's better.